engine/class_modules/priest/sc_priest.cpp

```diff
@@ -2198,21 +2198,30 @@ int priest_t::shadow_weaving_active_dots( const player_t* target ) const
       const dot_t* vt  = td->dots.vampiric_touch;
       const dot_t* dp  = td->dots.devouring_plague;
 
-      dots = swp->is_ticking() + vt->is_ticking() + dp->is_ticking();
+      // You get mastery benefit for a DoT as if it was active, if you are actively putting that DoT up
+      // So to get the mastery benefit you either have that DoT ticking, or you are casting it
+      const spell_data_t* swp_spell = find_class_spell( "Shadow Word: Pain" );
```

Comment:
definitely do not call find_class_spell here "during combat".

use some spell_data_t* variables stored in priest_t.